torchgen/api/types_base.py

```diff
@@ -0,0 +1,187 @@
+# (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+from abc import ABC
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional, Union
+from torchgen.model import (
+    Argument,
+    SelfArgument,
+    TensorOptionsArguments,
+)
```

Comment:
There should be some explanation at the top of this file explaining what the distinction between types_base.py and types.py is